### ISSUE TITLE
Add `cancelCallback` option like `successCallback`

### ISSUE THREAD

**Issue by terales** [yiisoft/yii2-authclient#196]
Hello, thanks a lot for this extension! It helps us a lot to connect to external auth providers securely and elegantly. However, I would suggest adding one minor feature.

### What steps will reproduce the problem?

1. Configure `yii\authclient\AuthAction` as described in official examples
1. Try to configure it 
   * to redirect back to the previous page,
   * to log cancelation,
   * to display some error message to a user that auth with external client won't work without access granted.

### What's expected?

All configuration is stored in one place inside controller

```php
// in custom AccountController->actions()
<?php
'create-external' => [
	'class' => \yii\authclient\AuthAction::class,
	'successCallback' => [$this, 'createFromExternal'],
	'cancelCallback' => function($client) { // any callable could be here
		Yii::$app->getSession()->setFlash('warning', 'You have to allow access to sign up with external source');
		return $this->redirect(['account/create']);
      },
],
```

### What do you get instead?

Configuration is split between several actions in the controller no matter how you would implement it:

<details>
<summary>1. Prepare a separate action just to add an error message — +one unnecessary redirect</summary>

```php
<?php
// in custom AccountController->actions()
'create-external' => [
	'class' => \yii\authclient\AuthAction::class,
	'successCallback' => [$this, 'createFromExternal'],
	'cancelUrl' => Url::to(['account/create-from-external-canceled']),
],

// in custom AccountController
public function createFromExternalCanceled($errorCode = '')
{
	Yii::$app->getSession()->setFlash('warning', 'You have to allow access to sign up with external source');
	return $this->redirect(['account/create']);
}
```

</details><br>

<details>
<summary>2. Set error code GET parameter with <code>cancelUrl</code> param — +one unnecessary GET param</summary>

```php
<?php
// in custom AccountController->actions()
'create-external' => [
	'class' => \yii\authclient\AuthAction::class,
	'successCallback' => [$this, 'createFromExternal'],
	'cancelUrl' => Url::to(['account/create', 'errorCode' => 'externalAuthCanceled']),
],

// in custom AccountController
public function actionCreate($errorCode = '')
{
	// …
	// code for sign up form validation and account creation…
	// …
	if ($errorCode === 'externalAuthCanceled')
		Yii::$app->getSession()->setFlash('warning', 'You have to allow access to sign up with external source');

	return $this->render('some-from');
}

```

</details>

### Additional info

I would love to prepare a PR, but I want to ask for your approval of the idea before diving into code.

| Q                       | A
| ----------------------- | ---
| Yii version             | 2.0.13.1
| Yii Auth Client version | 2.1.4
| Yii HTTP Client version | 2.0.5
| PHP version             | 7.2.1
| Operating system        | Windows 10 x64

**Comment by klimov-paul** [comment]
Resolved by commit 0a7b2a7bdd85f7f5c8481dcc72635ca18d8b3681

**Comment by terales** [comment]
Thanks a lot!

Do you have any other ideas to implement before `2.1.5` release?
